ChessVariantsLogic/Movement/IPattern.cs

```diff
@@ -1,9 +1,11 @@
+using System;
+
 namespace ChessVariantsLogic;
 
 /// <summary>
 /// An interface for different types of movement patterns.
 /// </summary>
-public interface IPattern
+public interface IPattern : IEquatable<IPattern>
```

Comment:
I couldn't find any place where the end result of this functionality is being used? Is it necessary?

There's some duplication going on within the implementors of `IPattern`. If we do need comparison, maybe it's better if we make `IPattern` an abstract class instead, which has the common behaviour.

Reply:
It needs to be able to compare patterns in order to remove patterns from the list when creating a custom `MovementPattern` in the pieceeditor.